planner/core/common_plans.go

```diff
@@ -974,8 +974,14 @@ func IsPointUpdateByAutoCommit(ctx sessionctx.Context, p Plan) (bool, error) {
 	if !ok {
 		return false, nil
 	}
-	if _, isFastSel := updPlan.SelectPlan.(*PointGetPlan); isFastSel {
-		return true, nil
+	if pointSel, isFastSel := updPlan.SelectPlan.(*PointGetPlan); isFastSel {
+		ok, err = IsPointGetWithPKOrUniqueKeyByAutoCommit(ctx, pointSel)
```

Comment:
why add it?

Reply:
there's one situation,    pointGet using unique key,  the plan cached and then another ddl do sth like `alter table drop index uniqueIndex`,   the cached plan will be invaliDated in `optimizePrepardPlan`,  but former `IsCachedExecOk` will return ok. Here is another change https://github.com/pingcap/tidb/pull/12486/files#diff-928cb614ae092ed4c6f349f4c4c991e4R1196,
I'm not quite sure if we should keep which one of  these check @jackysp @coocood

Reply:
Can we make `IsCachedExecOk` return not ok for this situation?